causalpy/tests/test_integration_pymc_examples.py

```diff
@@ -725,3 +725,283 @@ def test_inverse_prop(mock_pymc_sample):
+
+        model = cp.pymc_models.WeightedSumFitter(sample_kwargs=sample_kwargs)
+
+        sc = cp.SyntheticControl(
+            data=df,
+            treatment_time=treatment_time,
+            control_units=control_units,
+            treated_units=treated_units,
+            model=model,
+        )
+
+        # Test plotting - should work for each treated unit individually
+        for unit in treated_units:
+            fig, ax = sc.plot(treated_unit=unit)
+            assert isinstance(fig, plt.Figure)
+            assert isinstance(ax, np.ndarray) and all(
+                isinstance(item, plt.Axes) for item in ax
+            )
+
+        # Test default plotting (first unit)
+        fig, ax = sc.plot()
+        assert isinstance(fig, plt.Figure)
+        assert isinstance(ax, np.ndarray) and all(
+            isinstance(item, plt.Axes) for item in ax
+        )
+
+    @pytest.mark.integration
+    def test_multi_unit_plot_data(self, multi_unit_sc_data):
+        """Test that plot data generation works with multiple treated units."""
+        df, treatment_time, control_units, treated_units = multi_unit_sc_data
+
+        model = cp.pymc_models.WeightedSumFitter(sample_kwargs=sample_kwargs)
+
+        sc = cp.SyntheticControl(
+            data=df,
+            treatment_time=treatment_time,
+            control_units=control_units,
+            treated_units=treated_units,
+            model=model,
+        )
+
+        # Test plot data generation for each treated unit
+        for unit in treated_units:
+            plot_data = sc.get_plot_data(treated_unit=unit)
+            assert isinstance(plot_data, pd.DataFrame)
+
+            # Check expected columns
+            expected_columns = [
+                "prediction",
+                "pred_hdi_lower_94",
+                "pred_hdi_upper_94",
+                "impact",
+                "impact_hdi_lower_94",
+                "impact_hdi_upper_94",
+            ]
+            assert set(expected_columns).issubset(set(plot_data.columns))
+
+        # Test default plot data (first unit)
+        plot_data = sc.get_plot_data()
+        assert isinstance(plot_data, pd.DataFrame)
+
+    @pytest.mark.integration
+    def test_multi_unit_plotting_invalid_unit(self, multi_unit_sc_data):
+        """Test that plotting with invalid treated unit raises appropriate errors."""
+        df, treatment_time, control_units, treated_units = multi_unit_sc_data
+
+        model = cp.pymc_models.WeightedSumFitter(sample_kwargs=sample_kwargs)
+
+        sc = cp.SyntheticControl(
+            data=df,
+            treatment_time=treatment_time,
+            control_units=control_units,
+            treated_units=treated_units,
+            model=model,
+        )
+
+        # Test that invalid treated unit name is handled gracefully
+        # Note: Current implementation may not raise ValueError, so we test default behavior
+        try:
+            sc.plot(treated_unit="invalid_unit")
```

Comment:
I noticed when adding a test earlier that it asked for closing plots after opening
<img width="726" height="131" alt="image" src="https://github.com/user-attachments/assets/25c59581-8db0-44a4-aa07-4c635b627ecd" />